proto-util/src/main/java/games/mythical/proto_util/ProtoUtil.java

```diff
@@ -40,369 +41,416 @@
+            if (StringUtils.isNotBlank((String)sourceValue)) {
+                return JsonHelper.toStringObjectMap((String)sourceValue);
+            }
+            return Collections.emptyMap();
         }
-      }
-      return builder;
-    } catch (IntrospectionException | InvocationTargetException | NoSuchMethodException | IllegalAccessException ex) {
-      final var msg = String.format("Unable to convert %s to %s", source.getClass().getSimpleName(), resultClass.getSimpleName());
-      log.error(msg, ex);
-      throw new RuntimeException(msg, ex);
-    }
-  }
-
-  private static <T, R> R convert(final T source, final Class<R> resultClass, final boolean toProto) {
-    final var builder = convertToBuilder(source, resultClass, toProto);
-    try {
-      final var buildMethod = builder.getClass().getMethod(BUILD_METHOD_NAME);
-      //noinspection unchecked
-      return (R) buildMethod.invoke(builder);
-    } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException ex) {
-      final var msg = String.format("Unable to convert %s to %s", source.getClass().getSimpleName(), resultClass.getSimpleName());
-      log.error(msg, ex);
-      throw new RuntimeException(msg, ex);
-    }
-  }
 
-  private static <R> Object createBuilder(final Class<R> resultClass, final boolean toProto)
-      throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
-    final var builderCreator = resultClass.getMethod(toProto ? PROTO_BUILDER_NAME : LOMBOK_BUILDER_NAME);
-    return builderCreator.invoke(null);
-  }
+        // Note: The proto's type might not match what is expected. Strings are objects, Enums are ints, etc...
+        final var targetClass = targetField.getType();
+        // Catch Enum to Enum cases before turning the Enum into a String
+        if ((sourceValue.getClass() == targetClass) || (sourceValue instanceof ProtocolMessageEnum
+            && targetClass == Integer.TYPE)) {
+            return sourceValue;
+        }
+        if (sourceValue.getClass().isEnum()) {
+            return ((Enum<?>)sourceValue).name();
+        }
 
-  private static Optional<Field> getTargetField(final Field srcField, final Class<?> targetClass, final boolean isProto) {
-    if (isProto) {
-      return getTargetFieldInProto(srcField, targetClass);
-    }
-    return getTargetFieldInDto(srcField, targetClass);
-  }
-
-  private static Optional<Field> getTargetFieldInDto(final Field srcField, final Class<?> targetClass) {
-    final var name = stripUnderscore(srcField.getName());
-    final var fields = targetClass.getDeclaredFields();
-    for (final var field : fields) {
-      final var anno = field.getAnnotation(ProtoField.class);
-      if (anno != null) {
-        final var annotationValue = anno.value();
-        if (name.equals(annotationValue)) {
-          return Optional.of(field);
+        for (final var transformer : transformers) {
```

Comment:
wonder if there's a clean map way of doing this.

Reply:
Could do a find first and then return the transform if one is found, but I'm not sure that would be cleaner.